### ISSUE TITLE
Records constructor requires adjust_ratios absolute path

### ISSUE THREAD

**Issue by andersonfrailey** [open-source-economics/Tax-Calculator#2120]
This PR aims to report a potential bug in taxcalc when dealing with alternative adjustment ratios inputs. An example can be seen in [this notebook](https://github.com/andersonfrailey/Notebook-Uploads/blob/master/TaxCalcBug.ipynb).

When I was testing the updated data in TaxData PR [#289](https://github.com/open-source-economics/taxdata/pull/289), I tried to pass an alternative set of PUF ratios into the records class that included ratios up to 2028. However, when I examined the `ADJ` attribute of the records class it only included ratios up to 2027 (cell 4 in the notebook).

As can be seen in cells 5 and 6 of the notebook, the file I passed as an argument has ratios for up to 2028. And as can be seen in the last cell of the notebook, the weights go to 2028 so this issue only seems to be affecting the ratios.

cc @martinholmer

**Comment by martinholmer** [comment]
@andersonfrailey,  When I try to replicate your notebook results in a Python script run from the command line, I am not seeing the problem you encountered.  In fact, everything is working exactly the way we would expect.

While I'm not sure, I suspect that your use of a relative (rather than absolute) path to the new `puf_ratios.csv` file while using the taxcalc package caused taxcalc to not find the new ratios file and hence (as a backstop) it read in the `puf_ratios.csv` file embedded in the taxcalc package (which definitely has ratios only through 2027).

The rest of this comment shows what I did in a failed attempt to replicate the problem you reported.

I'm working in a directory where the Tax-Calculator source code is not visible and I installed the package for the 0.22.2 release of Tax-Calculator just as you did.   I have the old and new versions of the `puf_ratios.csv` file in that working directory along with the `puf.csv` file.  Here is the script I run (`python ratios.py`) once with OLD_RATIOS set to True and once with it set to False.
```
import os
import taxcalc as tc

OLD_RATIOS = True

print('Tax-Calculator', tc.__version__)
print('OLD_RATIOS=', OLD_RATIOS)
THIS_DIR = os.path.abspath(os.path.dirname(__file__))
if OLD_RATIOS:
    ratios_filename = os.path.join(THIS_DIR, 'puf_ratios.csv')
else:
    ratios_filename = os.path.join(THIS_DIR, 'puf_ratios-new.csv')
print('ratios_filename=', ratios_filename)

rec = tc.Records(adjust_ratios=ratios_filename)
print('rec.ADJ.columns=\n', rec.ADJ.columns)
print('rec.ADJ.shape=', rec.ADJ.shape)
```
And here is the output from that script: first when OLD_RATIOS is True:
```
Tax-Calculator 0.22.2
OLD_RATIOS= True
ratios_filename= /Users/mrh/work/PSL/taxcalc/puf_ratios.csv
rec.ADJ.columns=
 Index(['INT2011', 'INT2012', 'INT2013', 'INT2014', 'INT2015', 'INT2016',
       'INT2017', 'INT2018', 'INT2019', 'INT2020', 'INT2021', 'INT2022',
       'INT2023', 'INT2024', 'INT2025', 'INT2026', 'INT2027'],
      dtype='object', name='agi_bin')
rec.ADJ.shape= (19, 17)
```
And here is the output when OLD_RATIOS is False:
```
Tax-Calculator 0.22.2
OLD_RATIOS= False
ratios_filename= /Users/mrh/work/PSL/taxcalc/puf_ratios-new.csv
rec.ADJ.columns=
 Index(['INT2011', 'INT2012', 'INT2013', 'INT2014', 'INT2015', 'INT2016',
       'INT2017', 'INT2018', 'INT2019', 'INT2020', 'INT2021', 'INT2022',
       'INT2023', 'INT2024', 'INT2025', 'INT2026', 'INT2027', 'INT2028'],
      dtype='object', name='agi_bin')
rec.ADJ.shape= (19, 18)
```
So, it seems as if everything is working as expected, right?

**Comment by martinholmer** [comment]
@andersonfrailey,  Further experimentation with the script I described earlier in this issue shows that indeed there is a bug in the Records class constructor's handling of relative paths to a file containing adjustment ratios.  I hope to fix this bug soon in a future pull request.   Thanks for reporting this problem with Tax-Calculator.

**Comment by andersonfrailey** [comment]
Thanks for diagnosing this, @martinholmer. Let me know if there's any way I can help with the bug fix.

**Comment by martinholmer** [comment]
Issue #2120 has been (at least partially resolved) by pull request #2121.
